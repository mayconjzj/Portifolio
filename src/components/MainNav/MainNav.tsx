export const MainNav = () => {
  return (
    <nav>
      <ul>
        <li>Home</li>
        <li>Sobre mim</li>
        <li>Projetos</li>
        <li>Contato</li>
      </ul>
    </nav>
  );
};
