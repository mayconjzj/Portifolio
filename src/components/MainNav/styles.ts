import tw from 'tailwind-styled-components';

export const MainNav = tw.nav`
  md:flex
  hidden
`;
