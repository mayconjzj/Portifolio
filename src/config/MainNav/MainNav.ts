import { MainNavProps } from '@/models';

export const MainNavConfig: MainNavProps[] = [
  {
    title: 'Home',
    href: '#home'
  },
  {
    title: 'Sobre mim',
    href: '#about'
  },
  {
    title: 'Projetos',
    href: '#projects'
  },
  {
    title: 'Contato',
    href: '#contact'
  }
];
