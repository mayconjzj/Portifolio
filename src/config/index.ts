export * from './HomePage';
export * from './MainNav';
export * from './Logo';
