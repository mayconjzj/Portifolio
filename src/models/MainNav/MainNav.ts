export type MainNavProps = {
  title: string;
  href: string;
  icon?: React.ReactNode;
};
