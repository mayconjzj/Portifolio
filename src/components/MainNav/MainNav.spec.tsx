import { render } from '@testing-library/react';

import { MainNavConfig } from '@/config/MainNav';

import { MainNav } from './MainNav';

describe('MainNav', () => {
  it('should render the mainnav', () => {
    const { asFragment } = render(<MainNav items={MainNavConfig} />);
    expect(asFragment()).toBeTruthy();

    expect(asFragment()).toMatchSnapshot();
  });
});
