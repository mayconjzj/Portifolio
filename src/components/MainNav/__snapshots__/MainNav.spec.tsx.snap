// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainNav should render the mainnav 1`] = `
<div>
  <nav
    class=""
  >
    <ul
      class="flex gap-x-7"
    >
      <li>
        <a
          class="text-[#222] hover:text-blue-500 transition delay-150 cursor-pointer font-medium"
          isactive="false"
        >
          Next test
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
